src/backend/core/authentication/backends.py

```diff
@@ -90,7 +111,38 @@ def get_or_create_user(self, access_token, id_token, payload):
                 raise SuspiciousOperation(_("User account is disabled"))
             self.update_user_if_needed(user, claims)
         elif self.get_settings("OIDC_CREATE_USER", True):
-            user = User.objects.create(sub=sub, password="!", **claims)  # noqa: S106
+            user = self.create_user(claims)
+
+        # Data cleaning, to be removed when user organization is null=False
+        # or all users have an organization.
+        if not user.organization_id:
+            organization_registration_id = claims.get(
+                settings.OIDC_ORGANIZATION_REGISTRATION_ID_FIELD
+            )
+            domain = get_domain_from_email(email)
+            try:
+                organization, organization_created = Organization.objects.get_or_create(
+                    registration_id=organization_registration_id,
+                    domain=domain,
```

Comment:
Isn't it good enough if one of the fields matches? I'm thinking for example of a case where the organization ID was not transmitted by the OIDC provider and gets added suddenly.
What do we want to do in case where an organization exists with this domain but it was associated with another ID.

{EDIT] I realized further down the review that you handled it by overidding `get_or_create`. I find it confusing to override a well known method. Could it be a different name or is there a reason why you think it's better to override?  get_or_smart_create? ^^